oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java

```diff
@@ -0,0 +1,784 @@
+            LOG.debug("Got input stream for blob. identifier={} duration={}", key, (System.currentTimeMillis() - start));
+            return is;
+        }
+        catch (StorageException e) {
+            LOG.info("Error reading blob. identifier=%s", key);
+            throw new DataStoreException(String.format("Cannot read blob. identifier=%s", key), e);
+        }
+        catch (URISyntaxException e) {
+            LOG.debug("Error reading blob. identifier=%s", key);
+            throw new DataStoreException(String.format("Cannot read blob. identifier=%s", key), e);
+        } finally {
+            if (contextClassLoader != null) {
+                Thread.currentThread().setContextClassLoader(contextClassLoader);
+            }
+        }
+    }
+
+    @Override
+    public void write(DataIdentifier identifier, File file) throws DataStoreException {
+        if (null == identifier) {
+            throw new NullPointerException("identifier");
+        }
+        if (null == file) {
+            throw new NullPointerException("file");
+        }
+        String key = getKeyName(identifier);
+        long start = System.currentTimeMillis();
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+
+        try {
+            Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+
+            long len = file.length();
+            LOG.debug("Blob write started. identifier={} length={}", key, len);
+            CloudBlockBlob blob = getAzureContainer().getBlockBlobReference(key);
+            if (!blob.exists()) {
+                BlobRequestOptions options = new BlobRequestOptions();
+                options.setConcurrentRequestCount(concurrentRequestCount);
+                boolean useBufferedStream = len < BUFFERED_STREAM_THRESHHOLD;
+                final InputStream in = useBufferedStream  ? new BufferedInputStream(new FileInputStream(file)) : new FileInputStream(file);
+                try {
+                    blob.upload(in, len, null, options, null);
+                    LOG.debug("Blob created. identifier={} length={} duration={} buffered={}", key, len, (System.currentTimeMillis() - start), useBufferedStream);
+                } finally {
+                    in.close();
+                }
+                return;
+            }
+
+            blob.downloadAttributes();
+            if (blob.getProperties().getLength() != len) {
+                throw new DataStoreException("Length Collision. identifier=" + key +
+                                             " new length=" + len +
+                                             " old length=" + blob.getProperties().getLength());
+            }
+            LOG.trace("Blob already exists. identifier={} lastModified={}", key, blob.getProperties().getLastModified().getTime());
+            blob.startCopy(blob);
+            //TODO: better way of updating lastModified (use custom metadata?)
```

Comment:
I think it is worth checking into.

I'm not overly familiar with the history of S3Backend, but it was written prior to the API change for Oak 1.6, so at some point in time it needed to handle the ability to update the lastModified time.  If S3Backend is still doing this it might simply be because the code wasn't changed yet, and not because copying the blob is still required to hack the update of the lastModified time.